src/main/java/edu/cornell/kfs/tax/service/impl/PaymentReason1099BoxServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+package edu.cornell.kfs.tax.service.impl;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.coreservice.framework.parameter.ParameterService;
+
+import edu.cornell.kfs.tax.CUTaxConstants;
+import edu.cornell.kfs.tax.CUTaxConstants.Tax1099ParameterNames;
+import edu.cornell.kfs.tax.service.PaymentReason1099BoxService;
+
+public class PaymentReason1099BoxServiceImpl implements PaymentReason1099BoxService, Serializable {
```

Comment:
Using a service is one way of finding the parameter values. If you want to explore other options, you can try adding new fields to the Transaction1099Summary class, which already sets up various parameter-based immutable collections and maps based on parameter values. (Such "summary" objects are discarded after the batch job finishes, so they will be recreated with fresh parameter values on the next run.)